lib/pdk/module/build.rb

```diff
@@ -196,6 +196,13 @@ def ignored_files
 
                              PathSpec.new(data)
                            end
+
+        # Also ignore the target directory if it is in the module dir and not already ignored
+        if Find.find(@module_dir).include?(@target_dir) && !@ignored_files.match(File.basename(@target_dir) + '/')
+          @ignored_files = @ignored_files.add("\/#{File.basename(@target_dir)}\/")
```

Comment:
Should use `target_dir` reader method rather than referencing the instance variable directly (here and in the line above), makes it easier to mock when unit testing.